api/src/main/java/org/pmiops/workbench/reporting/ReportingTableService.java

```diff
@@ -0,0 +1,195 @@
+package org.pmiops.workbench.reporting;
+
+import jakarta.inject.Provider;
+import java.util.List;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.jdbc.ReportingQueryService;
+import org.pmiops.workbench.model.ReportingBase;
+import org.pmiops.workbench.model.ReportingCohort;
+import org.pmiops.workbench.model.ReportingDataset;
+import org.pmiops.workbench.model.ReportingDatasetCohort;
+import org.pmiops.workbench.model.ReportingDatasetConceptSet;
+import org.pmiops.workbench.model.ReportingDatasetDomainIdValue;
+import org.pmiops.workbench.model.ReportingInstitution;
+import org.pmiops.workbench.model.ReportingLeonardoAppUsage;
+import org.pmiops.workbench.model.ReportingNewUserSatisfactionSurvey;
+import org.pmiops.workbench.model.ReportingUser;
+import org.pmiops.workbench.model.ReportingUserGeneralDiscoverySource;
+import org.pmiops.workbench.model.ReportingUserPartnerDiscoverySource;
+import org.pmiops.workbench.model.ReportingWorkspace;
+import org.pmiops.workbench.model.ReportingWorkspaceFreeTierUsage;
+import org.pmiops.workbench.reporting.insertion.CohortColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.DatasetCohortColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.DatasetColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.DatasetConceptSetColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.DatasetDomainColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.InstitutionColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.LeonardoAppUsageColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.NewUserSatisfactionSurveyColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.UserColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.UserGeneralDiscoverySourceColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.UserPartnerDiscoverySourceColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.WorkspaceColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.WorkspaceFreeTierUsageColumnValueExtractor;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ReportingTableService {
```

Comment:
consolidate all reporting table constants and unique behavior here

Reply:
This doesn't follow our usual Service/Impl separation.  I didn't feel a real need for it.  But I can do that if you prefer.